ext/session/session.c

```diff
@@ -2379,7 +2373,7 @@ PHP_FUNCTION(session_create_id)
 	}
 
 	if (prefix && ZSTR_LEN(prefix)) {
-		if (php_session_valid_key(ZSTR_VAL(prefix)) == FAILURE) {
+		if (php_session_valid_key(prefix) == FAILURE) {
 			/* E_ERROR raised for security reason. */
 			php_error_docref(NULL, E_WARNING, "Prefix cannot contain special characters. Only the A-Z, a-z, 0-9, \"-\", and \",\" characters are allowed");
 			RETURN_FALSE;
```

Comment:
Also this comment is wrong. Not sure how this happened, and if the E_ERROR got demoted would be good to know why. Not that I will be much of an issue if we promote this to a ValueError.

Reply:
I tried to track it but even in 2020 there were incorrect comments: https://github.com/php/php-src/commit/f293e6b92003d02fe7db81861500e8ff8fb771c4#diff-22481788aee01357f8e628c9191632d127a78a2ea20cfbf9c3354cdc420d4ff2L2277

Reply:
Yeah, well that's less than ideal.